textile/overrides/work_order_hooks.py

```diff
@@ -141,3 +141,19 @@ def generator():
 def update_job_card_on_create(job_card):
 	pretreatment_order = frappe.db.get_value("Work Order", job_card.work_order, "pretreatment_order", cache=1)
 	job_card.pretreatment_order = pretreatment_order
+
+
+def work_order_list_query(user):
+	if not user:
+		user = frappe.session.user
+
+	# show only work orders to user those linked with only pretreatment orders/print orders based on the user roles
+	user_has_roles = frappe.get_roles(user)
+	conditions = []
+	if all([("Pretreatment Production User" not in user_has_roles), ("Pretreatment Sales User" not in user_has_roles)]):
+		conditions.append("(NOT(`tabWork Order`.pretreatment_order IS NOT NULL AND `tabWork Order`.pretreatment_order != ''))")
+	if all([("Print Production User" not in user_has_roles), ("Print Sales User" not in user_has_roles)]):
+		conditions.append("(NOT(`tabWork Order`.print_order IS NOT NULL AND `tabWork Order`.print_order != ''))")
```

Comment:
This is really poorly written .. needs to be simplified. We also need to add document permission hook which you missed in my comment https://frappeframework.com/docs/user/en/python-api/hooks#document-permissions